In this notebook we want to see that is there any relation between the deaths by COVID and tha tempreture.

In [1]:
# loading Lybreries
import pandas as pd
import numpy as np
import statsmodels.stats.weightstats

In [2]:
# laoding Data
data = pd.read_csv('Corona_Updated.csv')
data.head()

Country/Region          Last Update  Confirmed  Deaths  \
0              Mainland China  2020-03-10T15:13:05      67760    3024   
1                       Italy  2020-03-10T17:53:02      10149     631   
2  Iran (Islamic Republic of)  2020-03-10T19:13:20       8042     291   
3           Republic of Korea  2020-03-10T19:13:20       7513      54   
4                      France  2020-03-10T18:53:02       1784      33   

   Recovered  Latitude  Longitude  Temprature  Humidity  
0      47743   30.9756   112.2707        12.5        86  
1        724   43.0000    12.0000        12.9        64  
2       2731   32.0000    53.0000        11.9         9  
3        247   36.0000   128.0000         4.9        41  
4         12   47.0000     2.0000        11.9        93

In [3]:
# ckeck our data set for null cells
data.isnull().sum()

Country/Region    0
Last Update       0
Confirmed         0
Deaths            0
Recovered         0
Latitude          0
Longitude         0
Temprature        0
Humidity          0
dtype: int64

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 206 entries, 0 to 205
Data columns (total 9 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Country/Region  206 non-null    object 
 1   Last Update     206 non-null    object 
 2   Confirmed       206 non-null    int64  
 3   Deaths          206 non-null    int64  
 4   Recovered       206 non-null    int64  
 5   Latitude        206 non-null    float64
 6   Longitude       206 non-null    float64
 7   Temprature      206 non-null    float64
 8   Humidity        206 non-null    int64  
dtypes: float64(3), int64(4), object(2)
memory usage: 14.6+ KB


In [5]:
# check our features stitistics
data.describe()

Confirmed       Deaths     Recovered    Latitude   Longitude  \
count    206.000000   206.000000    206.000000  206.000000  206.000000   
mean     575.640777    20.689320    312.640777   31.184989   11.752030   
std     4822.697784   215.794845   3332.764713   21.305149   84.576291   
min        0.000000     0.000000      0.000000  -41.454500 -157.498300   
25%        3.000000     0.000000      0.000000   25.069200  -74.841325   
50%       12.000000     0.000000      0.000000   36.030550   15.234250   
75%       75.750000     1.000000      4.000000   43.870250  101.363375   
max    67760.000000  3024.000000  47743.000000   64.963100  174.886000   

       Temprature    Humidity  
count  206.000000  206.000000  
mean    12.161165   67.728155  
std     10.229763   21.780588  
min    -21.900000    6.000000  
25%      6.100000   55.000000  
50%     11.750000   73.000000  
75%     20.375000   84.000000  
max     33.100000   98.000000

the meadian for Temprature is 11.75 so we can catagorize the temprature data by being smaller or larger than 12

In [6]:
data['Temp_cat'] = data['Temprature'].apply(lambda x : 0 if x<12 else 1)

In [7]:
print('The number for locations with the Temprature Catagory of 0 (temprature smaller than 12): '   ,(data['Temp_cat'] == 0).sum())

The number for locations with the Temprature Catagory of 0 (temprature smaller than 12):  107


In [8]:
print('The number for locations with the Temprature Catagory of 1 (temprature larger than 12): '   ,(data['Temp_cat'] == 1).sum())

The number for locations with the Temprature Catagory of 1 (temprature larger than 12):  99


In [9]:
# define two datasets for temprature catagory of 0 and 1
data_A = data[data['Temp_cat']==0]['Deaths']
data_B = data[data['Temp_cat']==1]['Deaths']
print('data_A : number of Daeths for rows that have a Temp_cat of 0 ')
print('data_B : number of Daeths for rows that have a Temp_cat of 1 ')

data_A : number of Daeths for rows that have a Temp_cat of 0 
data_B : number of Daeths for rows that have a Temp_cat of 1 


Now it's the time for ABtesting
we want to campare the mean of data_A and the mean of data_B and see that are these means equals or not? so this is our hypothesis:

H0 : mean(data_A) = mean(data_B)

H1 : mean(data_A) != mean(data_B)

As we know the number of our data is 206 and we don't know the variance of society, but because n is larger than 30 we can use the Ztest

In [10]:
# AB testing our two datasets
statsmodels.stats.weightstats.ztest(data_A, data_B , alternative='two-sided')

(-1.1230490606590326, 0.26141665862176766)

The p_value for our ztest is 0.26, so by the error of 5 percent or 10 percent our H0 Hypothesis can't be rejected, so we can approve our H0 ,that means there is no difference between the mean of data_A and data_B

Finally we can say that the temprature doesn't have any influence on the number of deaths.